### PR TITLE
Scaffolding

### LINKED ISSUE #206: Scaffolding command for new built-in and custom rules
Something like:
```
regal new rule --type custom --category naming --title lower-case
```
Which would:
1. Create a `.regal` directory if none exists.
2. Create a `.regal/rules` directory if none exists.
3. Create a `.regal/rules/category/title.rego` file pre-populated with the basic required policy structure.
4. Create a `.regal/rules/category/title_test.rego` file pre-populated with the basic required policy test structure.

And something similar to the above for creating new built-in rules as well, where the new files are added to the Regal bundle instead, and where there's also an entry added for documentation in the docs directory.

### PR DESCRIPTION
**Pull request by anderseknert**
Add `regal new rule` command to allow quickly getting started working with custom rules, or builtin rules for Regal itself. See added docs for more details.

Fixes #206

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -27,7 +24,6 @@ func main() {
 	// so while this isn't pretty, we'll use a separate package to carry state from here. If you
 	// know of a better way of doing this, don't hesitate to fix this.
 	embeds.EmbedBundleFS = bundle
-	embeds.ASTSchema = schema
```
Comment (anderseknert) [inline main.go:30]: @charlieegan3 as you requested in a previous PR :) Moved to the embeds directory.

Comment (charlieegan3) [reply main.go:30]: 👏

File: docs/custom-rules.md [hunk 2]
```diff
@@ -21,6 +21,26 @@ linter rules might then look something like this:
 If you so prefer, custom rules may also be provided using the `--rules` option for `regal lint`, which may point either
 to a Rego file, or a directory containing Rego files and potentially data (JSON or YAML).
 
+## Creating a New Rule
```
Comment (charlieegan3) [inline docs/custom-rules.md:24]: Is it worth calling out this functionality for creating built-in rules in the readme so to guide new contributors?

**Review by charlieegan3 — APPROVED** [review]

File: docs/custom-rules.md (hunk 2 above)
Comment (anderseknert) [reply docs/custom-rules.md:24]: Hmm, I guess that'd be in the development.md file? I'll amend this with a note there.

**Pull request merged**